### PR TITLE
feat: Add updater for Current Transformer data

### PR DESCRIPTION
**Pull request by catsmanac**
Envoy Metered with consumption Current Transformers (CT) implemented report CT details in /ivp/meters/readings. For the consumption CT configured in `Solar with Load` this provides information on net-production (grid export) and net-consumption (grid-import). Net-consumption is also reported in /production as an increasing/decreasing total of import and export, CT readings however provides these as 2 increasing properties. [^1]

[^1]: variations between firmware release may exist.

This PR is step 3 of 3 to implement #69. It adds:

- update method implemented for meters updater
- CTMETERS supported feature added for meters probe
- Tests added for CT data and meters updater code coverage
- meters updater added to documentation index

### CODE REVIEW

File: src/pyenphase/updaters/meters.py [hunk 1]
```diff
@@ -111,12 +132,73 @@ async def probe(
...
+                for phase in range(self.phase_count if self.phase_count > 1 else 0):
+                    consumption: EnvoyMeterData | None = EnvoyMeterData.from_phase(
+                        meter, meters_status[index], phase
+                    )
+                    # exclude None phases that were expected but not actually in report
+                    if consumption:
+                        phase_consumption[PHASENAMES[phase]] = consumption
+
+                if len(phase_consumption) > 0:
+                    envoy_data.ctmeter_consumption_phases = phase_consumption
```
Comment (bdraco) [inline src/pyenphase/updaters/meters.py:204]:
```suggestion
        for index, meter in enumerate(meters_readings):
            # match meter identifier to one found during probe to identify production or consumption
            if meter["eid"] == self.production_meter_eid and self.production_meter_type:
                # if production meter was enabled (type known) store ctmeter production data
                envoy_data.ctmeter_production = EnvoyMeterData.from_api(
                    meter, meters_status[index]
                )
                # if more then 1 phase configured store ctmeter phase data
                envoy_data.ctmeter_production_phases = {
                    PHASENAMES[phase_idx]: production
                    for phase_idx in range(
                        self.phase_count if self.phase_count > 1 else 0
                    )
                    # exclude None phases that were expected but not actually in report
                    if (
                        production := EnvoyMeterData.from_phase(
                            meter, meters_status[index], phase_idx
                        )
                    )
                }
            # match meter identifier to one found during probe to identify production or consumption
            if (
                meter["eid"] == self.consumption_meter_eid
                and self.consumption_meter_type
            ):
                # if consumption meter was enabled (type known) store ctmeter consumption data
                envoy_data.ctmeter_consumption = EnvoyMeterData.from_api(
                    meter, meters_status[index]
                )
                # if more then 1 phase configured store ctmeter phase data
                envoy_data.ctmeter_consumption_phases = {
                    PHASENAMES[phase_idx]: consumption
                    for phase_idx in range(
                        self.phase_count if self.phase_count > 1 else 0
                    )
                    # exclude None phases that were expected but not actually in report
                    if (
                        consumption := EnvoyMeterData.from_phase(
                            meter, meters_status[index], phase_idx
                        )
                    )
                }
```

Comment (bdraco) [reply src/pyenphase/updaters/meters.py:204]: You'll likely have to adjust the tests as well for the empty dict instead of `None`

Comment (bdraco) [reply src/pyenphase/updaters/meters.py:204]: or you could write it as 
```python
                phase_consumption = {
                    PHASENAMES[phase_idx]: consumption
                    for phase_idx in range(
                        self.phase_count if self.phase_count > 1 else 0
                    )
                    # exclude None phases that were expected but not actually in report
                    if (
                        consumption := EnvoyMeterData.from_phase(
                            meter, meters_status[index], phase_idx
                        )
                    )
                }

                if phase_consumption:
                    envoy_data.ctmeter_consumption_phases = phase_consumption
```

But might be easier to have an empty dict

Comment (catsmanac) [reply src/pyenphase/updaters/meters.py]: Thanks, I've implemented the `for` changes. Opted to return `none` to be consistent with [Production](https://github.com/pyenphase/pyenphase/blob/main/src/pyenphase/updaters/production.py) updater that does the same for the phases.

**Comment by bdraco** [comment]
Tested, ok 👍

**Review by bdraco — APPROVED** [review]

**Comment by catsmanac** [comment]
Thanks @bdraco

**Pull request merged**